src/particle.cpp

```diff
@@ -203,11 +210,25 @@ void Particle::event_advance()
   double distance = std::min(boundary().distance, collision_distance());
 
   // Advance particle in space and time
+  // Short-term solution until the surface source is revised and we can use
+  // this->move_distance(distance)
```

Comment:
Why can't you use this?

Reply:
No worries. 

This was due to the bug you/we found in the **surface source test**. 
see: https://github.com/openmc-dev/openmc/pull/2454#discussion_r1263308672

As we use this operation (moving the particle) at several positions in the code it makes sense to functionalize it.

I thought we add it like this and someone (maybe you, cause I have no idea what is going on there...) will make the changes for the surface source / the surface source test and then we use `move_distance` also here. So its an interm solution.

What do you think?

Reply:
Ahh right, I totally forgot about the weirdness with that test. Thanks for reminding me :smile: I still don't have a good idea what's going on with it so, as you suggest, we'll just punt for now and leave the code as is.